content/oci/oci.go

```diff
@@ -375,6 +376,80 @@ func (s *Store) writeIndexFile() error {
 	return os.WriteFile(s.indexPath, indexJSON, 0666)
 }
 
+// Referrers lists the descriptors of image or artifact manifests directly
+// referencing the given manifest descriptor.
+//
+// fn is called on the referrer results. If artifactType is not empty, only
+// referrers of the same artifact type are fed to fn.
+//
+// Reference: https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc3/spec.md#listing-referrers
+func (s *Store) Referrers(ctx context.Context, desc ocispec.Descriptor, artifactType string, fn func(referrers []ocispec.Descriptor) error) error {
```

Comment:
This looks like a generic implementation that can work on any `ReadOnlyStorage` which support Predecessors. Can we have a generic implementation like [Tags](https://pkg.go.dev/oras.land/oras-go/v2@v2.3.1/registry#Tags)?

Reply:
Done a generic implementation, with an input parameter of `ReadOnlyGraphStorage`.